packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { gridContainerSizesSelector } from '../../../components/GridViewport';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
```

Comment:
I would be cautious with a use callback. Especially to make sure that the dependencies can't be eagerly created too often. The risk is to miss some scroll event during the unsubscribe/subscribe time. I think that it's fine for now. But in the future, we might need to use `useEventCallback` to fix a bug reported by developer. We will see. I have no idea if it will really happen.

Reply:
I think more testing will be needed. I can remove it as there isn't much of a performance gain of keeping it.

Reply:
Alright, so we can push it for later?

Reply:
Yeah, I was making a general comment. How the value of useCallback has most often nothing to do with performance but render update logics.